string.c

```diff
@@ -3554,19 +3554,34 @@ rb_str_casecmp_p(VALUE str1, VALUE str2)
 static VALUE
 str_casecmp_p(VALUE str1, VALUE str2)
 {
-    rb_encoding *enc;
-    VALUE folded_str1, folded_str2;
-    VALUE fold_opt = sym_fold;
-
-    enc = rb_enc_compatible(str1, str2);
-    if (!enc) {
-	return Qnil;
+    if (str1 == str2) return Qtrue;
+    if (!rb_enc_compatible(str1, str2)) {
+        return Qnil;
     }
 
-    folded_str1 = rb_str_downcase(1, &fold_opt, str1);
-    folded_str2 = rb_str_downcase(1, &fold_opt, str2);
+    if (ENC_CODERANGE(str1) == ENC_CODERANGE_7BIT && ENC_CODERANGE(str2) == ENC_CODERANGE_7BIT) {
+        if (!rb_str_comparable(str1, str2)) {
```

Comment:
This check seems unnecessary looking at what `rb_str_comparable` does.

Reply:
Hum, good point. It was useful before, but now that I restricted the optimization to `ENC_CODERANGE_7BIT`, it's kind og useless.